programs/blockLogTrim/blockLogTrim.cpp

```diff
@@ -0,0 +1,120 @@
+#include <iostream>
+#include <iomanip>
+#include <fstream>
+#include <unistd.h>         //truncate(char* path, u64 newLen)
+using namespace std;
+
+uint32_t reverseBytes(uint32_t x) {      //convert between big and little endian
+    uint32_t rval;
+    char* src= (char*)&x;
+    char* dst= (char*)&rval;
+    dst[0]= src[3];
+    dst[1]= src[2];
+    dst[2]= src[1];
+    dst[3]= src[0];
+    return rval;
+}
+
+char const* help=
+    "This utility truncates blocks.log and blocks.index after a given block number.\n"
+    "First arg is directory containing the two files then -b arg for last block to keep.\n"
+    "Example: blockLogTrim eos_data/blocks -b30111222\n";
+
+struct __attribute__((packed)) BlockHead {    //first 46 bytes of each block
+  uint32_t timestamp;
+  uint64_t prodname;
+  uint16_t confirmed;
+  uint32_t blknum;           //low 32 bits of previous blockid, is block number, is big endian
+  uint32_t previous;         //high 32 bits
+  uint64_t blkid[3];         //rest of 256 bit blockid for previous block

```

Comment:
If this has to be used the attribute names should match those in block_header.hpp